app/controllers/registration_commands_controller.rb

```diff
@@ -48,6 +50,14 @@ def host_groups_json
              .map { |hg| hg.as_json(methods: :inherited_operatingsystem_id) }
   end
 
+  # Select smart proxies that have enabled 'Templates' and 'Registration' features
+  def smart_proxies
+    with_templates    = Feature.find_by(name: 'Templates')&.smart_proxies.to_a
+    with_registration = Feature.find_by(name: 'Registration')&.smart_proxies.to_a
```

Comment:
I think `SmartProxy.with_feature('Templates')` is easier to read. My reasoning is that you read from left to right and then it's clear you get SmartProxy instances that happen to have a feature.
```suggestion
    with_templates    = SmartProxy.with_features('Templates').to_a
    with_registration =  SmartProxy.with_features('Registration').to_a
```